gymnasium/wrappers/to_array.py

```diff
@@ -0,0 +1,244 @@
+# This wrapper will convert array inputs from an Array API compatible framework A for the actions
+# to any other Array API compatible framework B for an underlying environment that is implemented
+# in framework B, then convert the return observations from framework B back to framework A.
+#
+# More precisely, the wrapper will work for any two frameworks that can be made compatible with the
+# `array-api-compat` package.
+#
+# See https://data-apis.org/array-api/latest/ for more information on the Array API standard, and
+# https://data-apis.org/array-api-compat/ for more information on the Array API compatibility layer.
+#
+# General structure for converting between types originally copied from
+# https://github.com/google/brax/blob/9d6b7ced2a13da0d074b5e9fbd3aad8311e26997/brax/io/torch.py
+# Under the Apache 2.0 license. Copyright is held by the authors
+
+"""Helper functions and wrapper class for converting between arbitrary Array API compatible frameworks and a target framework."""
+
+from __future__ import annotations
+
+import functools
+import numbers
+from collections import abc
+from types import ModuleType
+from typing import Any, Iterable, Mapping, SupportsFloat
+
+import gymnasium as gym
+from gymnasium.core import RenderFrame, WrapperActType, WrapperObsType
+from gymnasium.error import DependencyNotInstalled
+
+
+try:
+    from array_api_compat import array_namespace, is_array_api_obj, to_device
+
+except ImportError:
+    raise DependencyNotInstalled(
+        'Array API packages are not installed therefore cannot call `to_array`, run `pip install "gymnasium[array-api]"`'
+    )
+
```

Comment:
Could you add a numpy version check here

Reply:
Do you mean on top of pyproject.toml raising the numpy version to the required one? Yes, since Version has become part of the dependencies anyways that should not be an issue